### ISSUE TITLE
FR: Support for exposing a tailnet service using annotations through its name instead of IP

### ISSUE THREAD

**Issue by Shaked** [tailscale/tailscale#10280]
### What are you trying to do?

I have an automated GitOps setup which constructs the names of my cloud machines on different kubernetes clusters. I want to be able to generate a headless service automatically without the need to know about the TS IP addresses. So instead of using `tailscale.com/tailnet-ip: 100.68.29.93 // Tailscale IP address`, I'd like to do something like: 

```
apiVersion: v1
kind: Service
metadata:
  annotations:
    tailscale.com/hostname: ${AUTO_GENERATED_TS_HOSTNAME}
  name: nginx // service name
spec:
  externalName: unused // any value - will be overwritten by operator
  type: ExternalName
``` 

### How should we solve this?

This can be solved by running the same as `tailscale ip --4 $HOSTNAME` right before assigning the IP address to the generated TS proxy

### What is the impact of not solving this?

It will require us to manually add each server to our GitOps setup which losses the whole point of automating the setup. 

```[tasklist]
### Tasks
```

**Comment by irbekrm** [comment]
Thank you for opening the issue.

We would likely solve this by adding another annotation, similar to `tailscale.com/tailnet-ip` but for FQDNs i.e

```
apiVersion: v1
kind: Service
metadata:
  annotations:
    tailscale.com/tailnet-fqdn: <FQDN-of-tailnet-node>
  name: nginx // service name
spec:
  externalName: unused // any value - will be overwritten by operator
  type: ExternalName
```

**Comment by irbekrm** [comment]
Thank you for opening the issue @Shaked .

This was implemented in #10360 and the latest operator and proxy images with tags `unstable-v1.55.68` have this functionality.

See the description of https://github.com/tailscale/tailscale/pull/10360 for an example `Service` with the new annotation (we have not updated docs yet- waiting for a new minor tailscale release).

Do let us know if you have other suggestions with regards to enabling better GitOps flows.
